Refactored UA-Clarity-Tools.

EtienneThompson opened the pull request:
ua_clarity_tools.py
        - Made it so get_artifacts is only called when uri_only is not set
            to true.
        - Changed results argument to be default dict() when not specified
            rather than None and redundantly set it to dict() if not
            given another value.
    
test_ua_clarity_tools.py
        - Added check to make sure set container information is the same
            as what should've been scraped to the container_info test.

Fennersteel committed:
Made StepTools cache step_soup.

ua_clarity_tools.py
    - Fetched step_soup in __init__.
    - Removed API call to fetch step_details soup in each function
        and updated step_soup instances to self.step_soup.

Fennersteel committed:
Merge branch 'master' into Fennersteel/Single_Step_Soup

Fennersteel committed:
WIP

EtienneThompson committed:
Refactored UA-Clarity-Tools.

ua_clarity_tools.py
    - Made it so get_artifacts is only called when uri_only is not set
        to true.
    - Changed results argument to be default dict() when not specified
        rather than None and redundantly set it to dict() if not
        given another value.

test_ua_clarity_tools.py
    - Added check to make sure set container information is the same
        as what should've been scraped to the container_info test.

EtienneThompson committed:
Merge branch 'master' into Fennersteel/Container_Info

RyanJohannesBland commented on ua_clarity_tools/tests/test_ua_clarity_tools.py:
```diff
@@ -11,6 +11,7 @@
 from bs4 import BeautifulSoup
 from ua_clarity_tools import ua_clarity_tools
 from ua_clarity_tools import api_types
+import pdb
```
Please remove.

RyanJohannesBland commented on ua_clarity_tools/tests/test_ua_clarity_tools.py:
```diff
@@ -322,6 +323,51 @@ def test_get_artifact_map_one_input_one_output(self):
+        in_out_uris = self._harvest_art_uris("map")
+
+        all_uris = list(in_out_uris.keys())
+        all_uris.extend(list(in_out_uris.values()))
```
Are we sure this works? If these values contains lists (I.E. one input mapped to multiple outputs) this appends the list of outputs, not each individual outputs. Your assert using this doesn't fail, but that may be because the step we use to test is 1 to 1.

RyanJohannesBland commented on ua_clarity_tools/tests/test_ua_clarity_tools.py:
```diff
@@ -322,6 +323,51 @@ def test_get_artifact_map_one_input_one_output(self):
+            assert out_arts[0].location == expected_output.find(
+                "location").find("value").text
+
+        assert True
```
Remove please.

EtienneThompson replied:
Oh darn :(

RyanJohannesBland commented on ua_clarity_tools/ua_clarity_tools.py:
```diff
@@ -8,6 +8,7 @@
 from bs4 import BeautifulSoup, Tag
 from jinja2 import Template
 from ua_clarity_api import ua_clarity_api
+import pdb
```
Remove please.

EtienneThompson replied:
Oh darn :(

Fennersteel committed:
Fixed couple of errors.

ua_clarity_tools.py
    - Removed pdb import.

test_ua_clarity_tools
    - Removed pdb import.
    - Removed useless assert True statement.
    - Fixed issue if a step has multiple output artifacts mapped to a
        single input artifact, where the output artifacts wouldn't be
        properly processed.
